recipes/miniscript/all/conanfile.py

```diff
@@ -35,17 +58,39 @@ def configure(self):
     def layout(self):
         cmake_layout(self, src_folder="src")
 
+    def validate(self):
+        # miniscript doesn't declare export symbols with __declspec.
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} doesn't support msvc shared build.(yet)"
+            )
+
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
         tc = CMakeToolchain(self)
-        tc.variables["MINISCRIPT_SRC_DIR"] = self.source_folder.replace("\\", "/")
+        if Version(self.version) >= "1.6.2":
+            tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+        else:
+            tc.variables["MINISCRIPT_SRC_DIR"] = self.source_folder.replace("\\", "/")
```

Comment:
```suggestion
        if Version(self.version) < "1.6.2":
            tc.variables["MINISCRIPT_SRC_DIR"] = self.source_folder.replace("\\", "/")
```

I would not enforce shared libraries in Windows. Instead, I would recommend asking support from the upstream directly. Remember that `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` really fragile, it not only exposes all symbols, including those that should be private, but also lacks exposing public static variables, which causes runtime errors when being consumed (e.g. libcoro).